values.yaml

```diff
@@ -1,402 +1,134 @@
 services:
-  # services.nodeSelector -- Node selectors that will be applied to all Coder
-  # managed services.
-  # See: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector
+  # services.annotations -- A KV mapping of annotations. See:
+  # https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
+  annotations: {}
+  # services.tolerations -- Each element is a toleration object. See:
+  # https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/
+  tolerations: []
+  # services.nodeSelector -- See:
+  # https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector
   nodeSelector:
     kubernetes.io/os: linux
     kubernetes.io/arch: amd64
+  # services.type -- See the following for configurable types:
+  # https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types
+  type: "ClusterIP"
 
-# logging configures the logging format of Coder services. Specify one
-# or more targets for logging output. If you have multiple output
-# targets configured, coder will write output to multiple targets.
+# logging -- Configures the logging format and output of Coder.
 logging:
-  # logging.human -- Where to send logs that are formatted for readability by a
-  # human. Set to an empty string to disable.
+  # logging.human -- Location to send logs that are formatted for readability.
+  # Set to an empty string to disable.
   human: /dev/stderr
-  # logging.stackdriver -- Where to send logs that are formatted for Google
+  # logging.stackdriver -- Location to send logs that are formatted for Google
   # Stackdriver. Set to an empty string to disable.
   stackdriver: ""
-  # logging.json -- Where to send logs that are formatted as JSON. Set to an
-  # empty string to disable.
+  # logging.json -- Location to send logs that are formatted as JSON.
+  # Set to an empty string to disable.
   json: ""
 
-  # logging.splunk -- Coder can send logs directly to Splunk, in
-  # addition to file-based output, if these values are configured. The
-  # channel is optional, and this logging is disabled if either the
-  # URL and Token are set to the empty string.
+  # logging.splunk -- Coder can send logs directly to Splunk
+  # in addition to file-based output.
   splunk:
-    # logging.splunk.url -- The Splunk HEC collector endpoint.
+    # logging.splunk.url -- Splunk HEC collector endpoint.
     url: ""
-    # logging.splunk.token -- The Splunk HEC collector token.
+    # logging.splunk.token -- Splunk HEC collector token.
     token: ""
-    # logging.splunk_channel -- Optional. Specify the channel that you'd
-    # like to associate with all messages.
+    # logging.splunk_channel -- Optional. Specify the channel
+    # to associate messages with.
     channel: ""
-
-# serviceAccount configures the service account used for all Coder deployed
-# services.
-serviceAccount:
```

Comment:
Don't we still need a serviceAccount to run coderd itself? What service account do we use for this? Is this for control plane services such as coder-service, or are these used for workspaces?

Reply:
This should have an RFC, but I propose we remove the “built-in” provider in entirety.

Reply:
I think that all pods are associated with a service account. If we don't provide a service account name, then it will use the [`default` service account in the namespace](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/):

> A service account provides an identity for processes that run in a Pod.

and

> When you create a pod, if you do not specify a service account, it is automatically assigned the `default` service account in the same namespace.

This is true whether or not we mount the service account token into our pod (controlled with `automountServiceAccountToken: false` in the service account or Pod spec). The service account is associated with pull secrets, which we might need if the coderd image is in an air-gapped registry that requires password authentication.

Kube-Linter will emit a warning if pods use the default: https://docs.kubelinter.io/#/generated/checks?id=default-service-account - I'm not clear why this is a bad practice, but it seems to be :man_shrugging:

Reply:
I doubt we'll ever remove the built-in provider, we'll probably just migrate it to the new provider format by orphaning the resources or something.

Also, AFAIK default service account with `automountBlah: false` is fine.

Reply:
Do we need this option? Bit confused on this.

Reply:
Well, if you leave out the option then you'll end up with a service account mount (including token) for the default service account, which isn't authorized to do anything by default. I think it's best to set `automountServiceAccountToken: false`, yes